### ISSUE TITLE
FederatedStore should "save" graphs using the Gaffer Cache

### ISSUE THREAD

**Issue by GCHQDeveloper404** [gchq/Gaffer#1297]
A FederatedStore restart or re-initialisation would cause the federatedStore to loose all graphs added since the previous re-start/initialisation by users. This would require all graphs to be re-added all over again by users.

FederatedStore should be able to "save" all the graphs in scope and their auths.

This could also provide an Audit.

❓Thoughts required around using a cache to be storing internal details in the situation of crash, everything is automatically saved at all times.

**Comment by GCHQDeveloper404** [comment]
This has been brought up as unexpected behaviour by a user. User assumed that the properties in the library for the federated store would have updated to reflect the current state, so a reboot would return it back to how it was.

**Comment by GCHQDeveloper404** [comment]
# New Requirement
To avoid a Migration step being needed.
if the `gaffer.cache.service.class` property is not set or is null, then the FederatesStore should perform and work without any caching. An advanced user that questions why the graphs do not persist can then learn and add a persistent cache.

adding a non-persistent cache is pointless for default behaviour and would slow down performance with no benefit.

**Comment by GCHQDeveloper404** [comment]
blocked by #1431
and new requirement.

**Comment by GCHQDeveloper404** [comment]
In addition the whole federated across different joes issue.

**Comment by m55624** [comment]
In regards to the new requirement, this has been added.  If there is no cache class string set in the properties it will not use any cache and continue to work as it did before.  Also, the merging of Schema Id's and StoreProperties Id's has been addressed in https://github.com/gchq/Gaffer/issues/1431 and merged into this branch.  Tests have also been added on that branch.

**Comment by GCHQDeveloper404** [comment]
# MIGRATION

- Graphs are no longer added to FederatedStore via the property configuration.
- All graphs must be loaded via the AddGraphs methods/operations.
- The FederatedStore will be populated with any graphs that the optional cache is storing.

**Comment by GCHQDeveloper404** [comment]
blocked by #1472

**Comment by p013570** [comment]
Merged into develop.
